### PR TITLE
ci: Fix conventional commit title validation

### PR DESCRIPTION
**Pull request by spydon**
# Description

The conventional commit title validation wasn't running automatically when the contributor updated the title, this fixes so that it does.

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

## Related Issues

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

**Review by erickzanardo — APPROVED** [review]

**Review by st-pasha — APPROVED** [review]

**Comment by spydon** [comment]
Hmm, unclear why the title validation isn't starting...

File: .github/workflows/title-validation.yml [hunk 1]
```diff
@@ -1,14 +1,16 @@
-name: title-validation
-
+name: 'PR Title is Conventional'
 on:
```
Comment (spydon) [inline .github/workflows/title-validation.yml:2]:
```suggestion

on:
```

File: .github/workflows/title-validation.yml [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
+      - synchronize
 
 jobs:
-  validate-pr-title:
+  main:
     name: Validate PR title
     runs-on: ubuntu-latest
     steps:
-      - uses: amannn/action-semantic-pull-request@v4
+      - uses: amannn/action-semantic-pull-request@v4.5.0
```
Comment (spydon) [inline .github/workflows/title-validation.yml:14]:
```suggestion
      - uses: amannn/action-semantic-pull-request@v4
```

File: .github/workflows/title-validation.yml [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
-    types: [opened, reopened, synchronize]
+  pull_request_target:
+    types:
+      - opened
+      - edited
+      - synchronize
 
 jobs:
-  validate-pr-title:
+  main:
```
Comment (spydon) [inline .github/workflows/title-validation.yml:11]:
```suggestion
  validate-pr-title:
```

**Comment by spydon** [comment]
>>> [pull_request_target](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_target): This allows the action to be used in a fork-based workflow, where e.g. you want to accept pull requests in a public repository. In this case, the configuration from the main branch of your repository will be used for the check. This means that you need to have this configuration in the main branch for the action to run at all (e.g. it won't run within a PR that adds the action initially). Also if you change the configuration in a PR, the changes will not be reflected for the current PR – only subsequent ones after the changes are in the main branch.

So i'll just merge this and it should work

File: .github/workflows/title-validation.yml [hunk 4]
```diff
@@ -1,8 +1,11 @@
...
-  pull_request:
-    types: [opened, reopened, synchronize]
+  pull_request_target:
+    types:
+      - opened
+      - edited
+      - synchronize
 
 jobs:
   validate-pr-title:
```
Comment (spydon) [inline .github/workflows/title-validation.yml:11]:
```suggestion
  main:
```

**Pull request merged**